src/main/java/org/openstreetmap/atlas/checks/validation/tag/ConflictingCarAccessibilityCheck.java

```diff
@@ -0,0 +1,202 @@
+     * This function validates if given {@link AtlasObject} is valid for the check. A valid
+     * {@link AtlasObject} for this check must have the below properties: 1) Is an Edge 2) Has a
+     * highway tag 3) AccessTag for the object is not "no" 4) Has either `MotorVehicleTag` or
+     * `MotorcarTag` or `VehicleTag` and none of these are conditional tags 5) Is a master edge 6)
+     * An edge with the same OSM id has not already been flagged.
+     *
+     * @param object
+     *            the {@link AtlasObject} supplied by the Atlas-Checks framework for evaluation
+     * @return {@code true} if this object should be checked
+     */
+    @Override
+    public boolean validCheckForObject(final AtlasObject object)
+    {
+        // Checks if the object is an instance of Edge
+        return object instanceof Edge
+                // Make sure that AccessTag for the object is not "no"
+                && !AccessTag.isNo(object)
+                // Make sure that the object has either MotorVehicleTag or MotorcarTag or VehicleTag
+                // and does not contain any conditional restrictions on these tags
+                && ((object.getTag(MotorVehicleTag.KEY).isPresent()
+                        && !object.getTag(MotorVehicleTag.KEY + CONDITIONAL).isPresent())
+                        || (object.getTag(MotorcarTag.KEY).isPresent()
+                                && !object.getTag(MotorcarTag.KEY + CONDITIONAL).isPresent())
+                        || (object.getTag(VehicleTag.KEY).isPresent())
+                                && !object.getTag(VehicleTag.KEY + CONDITIONAL).isPresent())
+                // Make sure that only master edges are considered
+                && ((Edge) object).isMasterEdge()
+                // Make sure that the object has a Highway tag
+                && Validators.isOfType(object, HighwayTag.class, HighwayTag.values())
+                // Make sure that the object has not been previously flagged
+                && !this.isFlagged(object.getOsmIdentifier());
+    }
+
+    /**
+     * This is the actual function that will check to see whether the object needs to be flagged.
+     *
+     * @param object
+     *            the {@link AtlasObject} supplied by the Atlas-Checks framework for evaluation
+     * @return an optional {@link CheckFlag} object that contains the problem object and
+     *         instructions on how to fix it, or the reason the object was flagged
+     */
+    @Override
+    protected Optional<CheckFlag> flag(final AtlasObject object)
+    {
+        boolean isOverridingTag = false;
+        final Optional<Boolean> carAccessible = checkIfCarAccessible((Edge) object);
+        // If carAccessible is empty, do not consider the object for flagging
+        if (!carAccessible.isPresent())
+        {
+            return Optional.empty();
+        }
+        // Check if any one of the designated vehicle tag is present
+        if (this.designatedVehicleTagFilter.test(object))
```

Comment:
This can be simplified to `final boolean isOverridingTag = this.designatedVehicleTagFilter.test(object);`